src/value/mod.rs

```diff
@@ -167,6 +185,51 @@ pub(crate) fn mg_value_string(mg_value: *const bindings::mg_value) -> String {
     mg_string_to_string(c_str)
 }
 
+pub(crate) fn mg_value_naive_date(mg_value: *const bindings::mg_value) -> Option<NaiveDate> {
+    let c_date = unsafe { bindings::mg_value_date(mg_value) };
+    let c_delta_days = unsafe { bindings::mg_date_days(c_date) };
+    let epoch_date = NaiveDate::from_ymd(1970, 1, 1);
+    let delta_days = Duration::days(c_delta_days);
+    epoch_date.checked_add_signed(delta_days)
+}
+
+pub(crate) fn mg_value_naive_local_time(mg_value: *const bindings::mg_value) -> Option<NaiveTime> {
+    let nsec_in_sec = 1_000_000_000;
+    let u32_max = u32::MAX as i64;
+    let c_local_time = unsafe { bindings::mg_value_local_time(mg_value) };
+    let c_nanoseconds = unsafe { bindings::mg_local_time_nanoseconds(c_local_time) };
+    let seconds = c_nanoseconds / nsec_in_sec;
+    let nanoseconds = c_nanoseconds % nsec_in_sec;
+    if seconds > u32_max {
+        return None;
+    }
+    if nanoseconds > u32_max {
+        return None;
+    }
+    NaiveTime::from_num_seconds_from_midnight_opt(seconds as u32, nanoseconds as u32)
+}
+
+pub(crate) fn mg_value_naive_local_date_time(
+    mg_value: *const bindings::mg_value,
+) -> Option<NaiveDateTime> {
+    let u32_max = u32::MAX as i64;
```

Comment:
Do you really need to store this variable? Why not make the check directly on line 219? Also, you define it here but you use it a few lines later?